jormungandr/src/network/client/mod.rs

```diff
@@ -158,16 +158,17 @@ impl Progress {
 impl Client {
     fn process_block_event(&mut self, cx: &mut Context<'_>) -> Poll<Result<ProcessingOutcome, ()>> {
         use self::ProcessingOutcome::*;
-
         // Drive sending of a message to block task to clear the buffered
         // announcement before polling more events from the block subscription
         // stream.
+        let logger = self.logger().clone();
```

Comment:
Why `clone` here, a reference would work just as well?

Reply:
If I recall well in the `let mut block_sink = Pin::new(&mut self.block_sink);` call we take a mut ref to self which invalidates the others.

Reply:
Hmm; I'd expect the borrow checker to consider the member borrows separately.